src/app/task/task-list-item/task-list-item.component.ts

```diff
@@ -46,38 +58,110 @@ export class TaskListItemComponent implements OnChanges {
     const newTask: TaskInterface = {
       title: this.taskTitleInput,
       description: this.taskDescriptionInput,
-      status: StatusEnum.toDo,
-      comment: '',
+      comment: this.taskCommentInput,
+      priority: this.taskPriority,
+      status: this.taskStatus,
       id: this.task.id,
     }
     this.save.emit(newTask);
     this.addReadonly();
   }
 
-  onEditTask(): void  {
+  onEditTask(): void {
     this.editable = true;
-    if (!this.taskName || !this.taskDescription) {
+    if (!this.taskName || !this.taskDescription || !this.taskComment) {
       return;
     }
     this.taskName.nativeElement.removeAttribute('readonly');
     this.taskDescription.nativeElement.removeAttribute('readonly');
+    this.taskComment.nativeElement.removeAttribute('readonly');
   }
-  
-  undoChanges(): void  {
-    if (!this.taskName || !this.taskDescription) {
+
+  undoChanges(): void {
+    if (!this.taskName || !this.taskDescription || !this.taskComment) {
       return;
     }
     this.taskTitleInput = this.task.title;
     this.taskDescriptionInput = this.task.description;
+    this.taskCommentInput = this.task.comment
+    this.taskStatus = this.task.status;
+    this.taskPriority = this.task.priority;
     this.addReadonly();
   }
 
-  addReadonly(): void  {
+  addReadonly(): void {
     this.editable = false;
-    if (!this.taskName || !this.taskDescription) {
+    if (!this.taskName || !this.taskDescription || !this.taskComment) {
       return;
     }
     this.taskName.nativeElement.setAttribute('readonly', true);
     this.taskDescription.nativeElement.setAttribute('readonly', true);
+    this.taskComment.nativeElement.setAttribute('readonly', true);
+  }
+
+  getNextStatusIndex(): TaskStatusType {
+    let types: StatusEnum[] = Object.values(StatusEnum);
+    let indexOfNextStatus = types.indexOf(this.taskStatus!) + 1;
+    if (indexOfNextStatus > types.length - 1) {
+        indexOfNextStatus = 0;
+      }
+    return types[indexOfNextStatus];
+  }
+
+  changeStatus(): TaskStatusType | undefined {
+    if (!this.editable) return;
+    let state = this.getNextStatusIndex();
+    return this.taskStatus = state;
+  }
+
+  getBackgroundColor(): string | undefined {
+    let result: string | undefined;
+    switch (this.taskStatus) {
+      case StatusEnum.toDo:
+        result = 'inherit';
+        break;
+      case StatusEnum.inProgress:
+        result = 'violet';
+        break;
+      case StatusEnum.done:
+        result = 'green';
+        break;
+    }
+    return result;
+  }
+
+  get priorities() {
```

Comment:
add type of a value that method returns. Why do you need this method?

Reply:
I need this method that creates a list of priorities